go-controller/pkg/clustermanager/pod/allocator.go

```diff
@@ -131,6 +136,31 @@ func (a *PodAllocator) reconcile(old, new *corev1.Pod, releaseFromAllocator bool
 		pod = new
 	}
 
+	ns, err := a.namespaceLister.Get(pod.Namespace)
+	if err != nil {
+		return err
+	}
+	activeNetworkName, hasAnActiveNetwork := ns.Annotations["k8s.ovn.org/active-network"]
```

Comment:
waiting on @tssurya 's helper to identify the namespace's user defined primary net.